File: mobile/tool/gen_l10n_variants.dart
import 'dart:io';

void main() {
  final base = Directory('lib/l10n');
  final sr = File('${base.path}/app_sr.arb').readAsStringSync();

  var bs = sr.replaceFirst('"@@locale": "sr"', '"@@locale": "bs"');
  bs = bs.replaceAll('Nesinhronizovani', 'Nesinhronizirani');
  bs = bs.replaceAll('Izmena pčelinjaka', 'Izmjena pčelinjaka');
  bs = bs.replaceAll('"edit": "Izmeni"', '"edit": "Izmijeni"');
  bs = bs.replaceAll('Izmeni u grupi', 'Izmijeni u grupi');
  File('${base.path}/app_bs.arb').writeAsStringSync(bs);

  final cnr = sr.replaceFirst('"@@locale": "sr"', '"@@locale": "cnr"');
  File('${base.path}/app_cnr.arb').writeAsStringSync(cnr);

  stdout.writeln(File('${base.path}/app_bs.arb').readAsStringSync().substring(0, 80));
  stdout.writeln('ok');
}

File: mobile/tool/gen_l10n_variants.py
from pathlib import Path

base = Path(r"C:\Users\djord\projects\pcelinjak\mobile\lib\l10n")
sr = (base / "app_sr.arb").read_text(encoding="utf-8")

bs = sr.replace('"@@locale": "sr"', '"@@locale": "bs"')
bs = bs.replace("Nesinhronizovani", "Nesinhronizirani")
bs = bs.replace("Izmena pčelinjaka", "Izmjena pčelinjaka")
bs = bs.replace('"edit": "Izmeni"', '"edit": "Izmijeni"')
bs = bs.replace("editInGroup\": \"Izmeni u grupi", "editInGroup\": \"Izmijeni u grupi")
bs = bs.replace("Izmeni košnicu", "Izmijeni košnicu")
bs = bs.replace("Izmeni trenutnu", "Izmijeni trenutnu")
bs = bs.replace("Izmeni maticu", "Izmijeni maticu")
bs = bs.replace("Izmeni (nije dozvoljeno)", "Izmijeni (nije dozvoljeno)")
(base / "app_bs.arb").write_text(bs, encoding="utf-8")

cnr = sr.replace('"@@locale": "sr"', '"@@locale": "cnr"')
(base / "app_cnr.arb").write_text(cnr, encoding="utf-8")

print((base / "app_bs.arb").read_text(encoding="utf-8")[:120])
print("ok")
